wordpress-mobile/WordPress-Android
# Quickstart focus point change to quick link ribbon

Fixes #16227 

This PR adds the functionality of the quick start focus point on the quick link ribbon if the quick-start card is visible on the Dashboard menu. 

| Quick start focus point in Quick Actions Card  | Quick start focus point in Quick Link Ribbon | 
|---|---|
| ![Screenshot_20220407_185934](https://user-images.githubusercontent.com/17463767/162210364-6ba6583b-55cb-416d-a155-b21238d768a0.png) | ![Screenshot_20220407_190009](https://user-images.githubusercontent.com/17463767/162210374-50174c3b-f907-412e-9f68-ce4221dffffe.png)| 

### Current Implementation 

Please find below the video capture which shows the quick start focus point shown in different scenarios. Due to the limit of mp4 size in GitHub readme, the preview is not shown in the description. Please visit the link 

1. When Tabs feature Disabled - [Video Capture](https://user-images.githubusercontent.com/17463767/162193078-bd9a4b50-392c-4a0a-8078-26142193bf13.mp4)
- All Quickstart focus points should be shown on the Quick Link action card

2. When the Tabs feature is enabled and the default tab is **Site Menu** - [Video Capture](https://user-images.githubusercontent.com/17463767/162192206-9373b7c5-4869-48fc-ae6e-73e754aea0e0.mp4)

- All Quickstart focus points should be shown in the Site Menu tab
- Quick start focus points for Edit your home Page, Review Pages and Check stats should be shown on Quick Link Action Card in the Site Menu Tab  


3.  When the Tabs feature is enabled and the default tab is Dashboard - [Video Capture](https://user-images.githubusercontent.com/17463767/162191361-1b29b136-9ea9-49b2-971e-99c7d042fc7d.mp4)
- Quick start focus points for Edit your home Page, Review Pages and Check stats should be shown on Quick Link Ribbon in the Dashboard Tab  
- Quick start focus points for Check your Site, Social Sharing and Explore plans should show an intermediary step in site menu and should show the site menu item 


### To test:

Test 1 
- Launch the app with a site having a quick start in progress 
- Go to My Site -> Me -> App Settings -> Debug settings.
- Turn off the MySiteDashboardTabsFeatureConfig flag under the Features in development section.
Restart the app.
- Logout/login and select "Show me around" when the quick start prompt is shown.
- Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt).
- Notice that for the following tasks - **Edit your home Page**, **Review Pages** and **Check stats**, the QuickStart focus point is shown in Quick links 

Test 2 & 3 - Prerequisites
- Go to My Site -> Me -> App Settings -> Debug settings.
- Enable `MySiteDashboardTabsFeatureConfig`
- Restart the app

   
Test 2 - When the default tab is `Site Menu` 
- Navigate to `My Site` -> `Me` -> `App Settings` 
- Tap `Initial Screen` and select `Site Menu` 
- Launch the app with a site having a quick start in progress 
- Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt).
- Notice that for the following tasks - **Edit your home Page**, **Review Pages** and **Check stats**, the QuickStart focus point is shown in the Quick links Actions Card. 
- Notice that for the quick start tasks  - **Check your Site, Social Sharing and Explore plans**, the Quickstart focus points are shown in the Site menu items as same as before. 

Test 3 - When the default tab is `Home`
- Navigate to `My Site` -> `Me` -> `App Settings` 
- Tap `Initial Screen` and select `Home` 
- Launch the app with a site having a quick start in progress 
- Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt).
- Notice that for the following tasks - **Edit your home Page**, **Review Pages** and **Check stats**, the QuickStart focus point is shown in the Quick links Ribbon Card. 
- Notice that for the quick start tasks  - **Check your Site, Social Sharing and Explore plans**, the Quickstart focus points is shown in the Site menu tab header first
- On clicking the Site menu tab header, the list gets scrolled to show the Site menu item corresponding to the task 

## Regression Notes
1. Potential unintended areas of impact
Quick start focus points not being shown properly 
Quick start not working properly 


2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and Unit testing

3. What automated tests I added (or what prevented me from doing so)
Updated tests in MySiteViewModelTest

PR submission checklist:
